packages/alfa-dom/src/matches.ts

```diff
@@ -430,12 +430,24 @@ function matchesPseudoClass(
 
     // https://drafts.csswg.org/css-scoping/#host-selector
     case "host":
+      // Do not allow prefix (e.g. "div:host")
+      if (root !== selector) {
+        return false;
+      }
+
+      if (options.treeContext === undefined) {
+        return false;
+      }
+
+      const isHostMatched =
+        selector.value === null ||
+        matches(element, context, selector.value, options, root);
+
       return (
-        root === selector &&
-        options.treeContext !== undefined &&
         isShadowRoot(options.treeContext) &&
         getParentElement(options.treeContext, context, { composed: true }) ===
-          element
+          element &&
+        isHostMatched
```

Comment:
There is something funky about these indents, isn't there?

Reply:
It. Is. Impossible to fix. 😭

Reply:
You can pull out `getParentElement(options.treeContext, context, { composed: true })` to a variable and reference that instead. [Prettier](https://github.com/prettier/prettier) is pretty particular on indents 🙈